pycqed/analysis_v2/residual_ZZ_analysis.py

```diff
@@ -0,0 +1,192 @@
+    damped_osc_mod.set_param_hint('amplitude',
+                                  value=0.5*(max(y)-min(y)),
+                                  min=min(y),
+                                  max=max(y),
+                                  vary=True)
+    damped_osc_mod.set_param_hint('tau',
+                                  value=x[1]*10,
+#                                   min=x[1],
+#                                   max=x[1]*1000,
+                                  vary=True)
+    damped_osc_mod.set_param_hint('exponential_offset',
+                                  value=average,
+                                  min=min(y),
+                                  max=max(y),
+                                  vary=True)
+    damped_osc_mod.set_param_hint('oscillation_offset',
+                                  value=0,
+                                  vary=False)
+    damped_osc_mod.set_param_hint('n',
+                                  value=1,
+                                  vary=False)
+    params = damped_osc_mod.make_params()
+
+    fit_res = damped_osc_mod.fit(data=y,
+                                 t=x,
+                                 params=params)
+    return fit_res
+
+def plot_crosstalk_matrix(crosstalk_matrix,
+                          qubit_labels, combinations=None, ax=None,
+                          valid_combinations=None, **kw):
+    crosstalk_matrix = np.copy(crosstalk_matrix)/1000
+    if ax is None:
+        figsize = np.array(np.shape(crosstalk_matrix))
+        f = plt.figure(figsize=figsize)
+        ax = f.add_axes([0.2, 0.2, 0.6, 0.6])
+    else:
+        f = ax.get_figure()
+    if combinations is None:
+        combinations = qubit_labels
+    if valid_combinations is None:
+        valid_combinations = combinations
+    
+    max_range = 500*np.ceil(np.nanmax(np.abs(crosstalk_matrix))/500)
+    min_range = -1*max_range
+    mat_plot = ax.matshow(crosstalk_matrix,
+                        cmap=cm.RdBu, clim=(-max_range, max_range))
+
+    caxr = f.add_axes([0.85, 0.25, 0.02, 0.5])
+    ax.figure.colorbar(mat_plot, ax=ax, cax=caxr, label='Frequency shift (kHz)')
+
+    rows, cols = np.shape(crosstalk_matrix)
+    for i in range(rows):
+        for j in range(cols):
+            c = crosstalk_matrix[i, j]
+            if c > 0.5*max_range or c < 0.5*min_range:
+                col = 'white'
+            else:
+                col = 'black'
+            text_filtered = '{:.0f}'.format(c)
+            if np.isnan(c):
+                text_filtered = '-'
+            ax.text(j, i, text_filtered,
+                    va='center', ha='center', color=col)
+
+    ax.set_xticklabels(valid_combinations)
+    ax.set_xticks(np.arange(len(valid_combinations)))
+    ax.xaxis.set_ticks_position('bottom')
+
+    ax.set_yticklabels(combinations)
+    ax.set_yticks(np.arange(len(combinations)))
+    ax.set_ylim(len(combinations)-.5, -.5)
+    ax.set_ylabel('Echo qubit')
```

Comment:
I don't understand what these labels mean. 
Echo qubit and control qubit have some specific meaning in the experiment but that is not clear here. It may still be a good label but it definitely needs a comment to explain it.